### ISSUE TITLE
More configuration options

### ISSUE THREAD

**Issue by meskallito** [apache/solr-operator#55]
Hi there,

We are considering using Bloomberg operator for our SOLR cluster but there are a few things we need to configure that the operator doesn't provide for us:

- Support of configuration for security contexts on stateful sets 
- Custom annotations on services and stateful sets
- Adding our data volumes to stateful sets
- Setting policy name on collections

**Comment by HoustonPutman** [comment]
Hey @meskallito thanks for the input. We can definitely look at adding these things, they seem reasonable to me at a first-glance.

We are no longer blocked on #38 , so if you are interested in creating a PR with one or more of these added, we will be happy to look over and review it! And hopefully eventually we get all of these features added.

**Comment by HoustonPutman** [comment]
`Spec.customSolrKubeOptions.podOptions` and `Spec.customSolrKubeOptions.statefulSetOptions` have been added.

This allows for custom:
- Security Context
- Volumes
- Annotations on statefulSets/pods

What we still need:
- Service annotations
- Setting policy name on collections.

**Comment by HoustonPutman** [comment]
Service annotations have been added in `Spec.customSolrKubeOptions.commonServiceOptions`, `Spec.customSolrKubeOptions.headlessServiceOptions` and `Spec.customSolrKubeOptions.nodeServiceOptions`.

Last request is setting the policy name on a collection.

**Comment by HoustonPutman** [comment]
Since the SolrCollection CRD has been removed, this issue can be closed. All features on the SolrCloud CRD have been added.

**Comment by pareekdevanshu** [comment]
@HoustonPutman I was looking for an option to set securityContext for solr-cloud containers but couldnt find any option in the CRD to do that. I only see option to set securityContext for `sidecarContainers` and `initContainers`.
Am I missing something?

**Comment by HoustonPutman** [comment]
You can set the podSecurityContext through `customSolrKubeOptions.podOptions.podSecurityContext`

**Comment by pareekdevanshu** [comment]
@HoustonPutman Thanks for the reply, I wanted to `drop all capabilities in security context `  which can be done using securityContext :
```
securityContext:
  capabilities:
    drop:
      - ALL
     
``` 
can you please tell me how can we achieve the same using `customSolrKubeOptions.podOptions.podSecurityContext`? 
I couldnt find a way to mention the capabilities in `podSecurityContext`.

**Comment by HoustonPutman** [comment]
Ahh it looks like you can only set capabilities on the container security context. If you make a separate issue we can try to get that added.

**Comment by pareekdevanshu** [comment]
thanks, created this issue https://github.com/apache/solr-operator/issues/489
